tools/device_configuration_checker.py

```diff
@@ -345,4 +349,161 @@ def _run_tests_on_know_good_image(args, device):
         return (result, result_str)
 
     except Exception as error:
+        import traceback
+        traceback.print_exc()
+        logger.error(traceback.format_exc())
         return (False, "Image Test result: " + str(error))
+
+def create_work_directory(directory):
+    """
```

Comment:
I think creating a new class hierarchy for a single method is an overkill. If there was more functionality that needs to be specialized on a per-device basis, it would make sense.